In [ ]:
NAME: "Willem van der Meiden, 2014236"
COLLABORATOR: ""

# Assignment week 6

Mark Klik & Misja Mikkers


# Packages

We need the package `tidyverse`

In [1]:
library(tidyverse) 

Loading tidyverse: ggplot2
Loading tidyverse: tibble
Loading tidyverse: tidyr
Loading tidyverse: readr
Loading tidyverse: purrr
Loading tidyverse: dplyr
Conflicts with tidy packages ---------------------------------------------------
filter(): dplyr, stats
lag():    dplyr, stats


# Assignment 1

First, we generate some random data:

In [2]:
df1 <- as.data.frame(matrix(runif(20*2), ncol=20) )
head(df1)

V1        V2        V3        V4         V5        V6        V7       
1 0.6105973 0.6080213 0.1615040 0.53338003 0.1384484 0.3760077 0.5476406
2 0.4737759 0.7420444 0.2732995 0.08493023 0.8850959 0.5268705 0.2251633
  V8          V9        V10       V11       V12       V13       V14      
1 0.001384433 0.9152851 0.8971787 0.3673973 0.8541916 0.4242407 0.5332286
2 0.171484403 0.3258852 0.9184452 0.4779831 0.7167704 0.3653996 0.9097670
  V15       V16       V17       V18       V19        V20      
1 0.7307096 0.9839019 0.6510438 0.4135710 0.33519566 0.1728243
2 0.1988637 0.5494612 0.3218872 0.2163371 0.04978751 0.1183385

_df1_ is a wide dataframe. The goal of the assignment is to change _df1_ into a long dataframe _df2_. And show the first rows of the dataframe using `head(df2)`

The dataframe _df2_ should look like this:

| Key | Value      |
|-----|------------|
| V1  | 0.09413864 |
| V1  | 0.69782526 |
| V2  | 0.23460454 |
| V2  | 0.51520514 |
| V3  | 0.63181844 |
| V3  | 0.22807211 |

In [11]:
df2 <- gather(df1, Key, Value, V1:V20) %>%
    arrange()

head(df2)

Key Value    
1 V1  0.6105973
2 V1  0.4737759
3 V2  0.6080213
4 V2  0.7420444
5 V3  0.1615040
6 V3  0.2732995

## Assignment 2

We will generate some data:

In [12]:
df3 <- as.data.frame(read.table(header=TRUE, text='
      id        V_1       V_2                
          1  42.467573  -87.810047     
          2  42.049148  -88.273029
          3  39.110539  -90.324080
                       '))

head(df3)

id V_1      V_2      
1 1  42.46757 -87.81005
2 2  42.04915 -88.27303
3 3  39.11054 -90.32408

In [13]:
df4 <- as.data.frame(read.table(header=TRUE, text='
     V_3            V_4         id                 
      one         1h            1
      two         1h            1
      three       8h            2
      four        1h            4
'))

head(df4)

V_3   V_4 id
1 one   1h  1 
2 two   1h  1 
3 three 8h  2 
4 four  1h  4


You will have to merge these files, returning all rows and all columns from both _df3_ and _df5_. Where there are not matching values, return _NA_ for the missings. 

This should be the end result:

| id | V_1      | V_2       | V_3   | V_4 |
|----|----------|-----------|-------|-----|
| 1  | 42.46757 | -87.81005 | one   | 1h  |
| 1  | 42.46757 | -87.81005 | two   | 1h  |
| 2  | 42.04915 | -88.27303 | three | 8h  |
| 3  | 39.11054 | -90.32408 | NA    | NA  |
| 4  | NA       | NA        | four  | 1h  |



In [14]:
df5 <- full_join(df3, df4, by = 'id')
df5

id V_1      V_2       V_3   V_4
1 1  42.46757 -87.81005 one   1h 
2 1  42.46757 -87.81005 two   1h 
3 2  42.04915 -88.27303 three 8h 
4 3  39.11054 -90.32408 NA    NA 
5 4        NA        NA four  1h

End of notebook